### PR TITLE
Import Repository

### LINKED ISSUE #8: Import Selected Repository
### Description
The user should be presented with a drop-down list of all their public repositories. Once they select a given repository, a second drop-down list should be populated with a list of the repositories branches (defaulted to the repository's default branch i.e. `master`/`main`).

Once the user has selected both the repository and branch, add a database entry for the given data.

This should take place on the `repositories/import` URL.

**Collection:** `repositories`
**Document ID:**  Let Firestore generate this.
**Document Data:**
```ts
{
    created: Date; // The date at which the document is created
    git_url: String; // The git URL for the repository
    branch: String; // The branch selected
    commit: String; // The latest commit on the selected branch at time of import
    user: string; // The user's Firebase ID
    reviewers: string[]; // Empty array for the time being
}
```

### UI Wireframe
image

Population of the branch drop-down should happen on selection of the repository drop-down.
Creation of the database entry should happen on press of "Next".
 
### User Story
**As a** user
**I want to** be able to import a repository
**So that** I can get the code reviewed

### Notes
- Firebase's [`serverTimestamp()`](https://cloud.google.com/firestore/docs/manage-data/add-data#server_timestamp) method should be used for the `created` field
- GitHub API Documentation:
  - [List public repositories for a given user](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-repositories-for-a-user)
  - [List branches for a given repository](https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#list-branches)

### PR DESCRIPTION
**Pull request by LamboCreeper**
Resolves #8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,10 +3,18 @@
 ## Getting Started
 1. Set the correct Node version via `nvm use`
 2. Start the Firebase Emulators via `npm run dev`
-3. In another Terminal, start the React development server by running `npm start` within the `hosting` directory
-   - In order to authenticate locally, you will need to use `127.0.0.1` rather than `localhost`
+3. In another Terminal:
+   1. Navigate into the `hosting` directory
+   2. Run `npm i`
+   3. Create a `.env` file based off of `.env.sample`
+      - You can get a GitHub token from [here](https://github.com/settings/tokens)
```
Comment (saramaebee) [inline README.md]: Maybe explain which scopes are required and/or recommended

File: README.md [hunk 2]
```diff
@@ -3,10 +3,18 @@
 ## Getting Started
 1. Set the correct Node version via `nvm use`
 2. Start the Firebase Emulators via `npm run dev`
-3. In another Terminal, start the React development server by running `npm start` within the `hosting` directory
-   - In order to authenticate locally, you will need to use `127.0.0.1` rather than `localhost`
+3. In another Terminal:
+   1. Navigate into the `hosting` directory
+   2. Run `npm i`
```
Comment (saramaebee) [inline README.md:8]: How come you mention `npm i` here, but not for the root directory?

**Review by saramaebee — APPROVED** [review]
Aside from those two notes on the Readme, everything works as expected

File: README.md (hunk 2 above)
Comment (LamboCreeper) [reply README.md:8]: There's no dependencies in the root directory - the Firebase CLI is triggered via `npx` so doesn't need to be pre-installed via `npm i`

**Pull request merged**